crates/codegen/runtime/cargo/src/runtime/cli/mod.rs

```diff
@@ -0,0 +1,65 @@
+use std::process::ExitCode;
+
+use clap::Subcommand;
+use semver::Version;
+
+pub mod commands;
+
+#[derive(Subcommand, Debug)]
+pub enum Commands {
+    /// Parses a source file, and outputs any syntax errors, or a JSON concrete syntax tree
+    Parse {
+        /// File path to the source file to parse
+        file_path: String,
+
+        /// The language version to use for parsing
+        #[arg(short, long)]
+        version: Version,
+
+        /// Print the concrete syntax tree as JSON
+        #[clap(long)]
+        json: bool,
+    },
+
+    /// Parses a source file and builds a graph executing the instructions from the builder file (*.msgb)
+    BuildGraph {
+        /// File path to the source file to parse
+        file_path: String,
+
+        /// The language version to use for parsing
+        #[arg(short, long)]
+        version: Version,
+
+        /// The graph buider (.msgb) file to use
+        msgb_path: String,
+
+        /// Print the graph as JSON
+        #[clap(long)]
+        json: bool,
+    },
+}
+
+impl Commands {
+    pub fn execute(self) -> ExitCode {
+        let command_result = match self {
+            Commands::Parse {
+                file_path,
+                version,
+                json,
+            } => commands::execute_parse_command(&file_path, version, json),
+            Commands::BuildGraph {
```

Comment:
My understanding is that the new command added to `slang_solidity` CLI would take a source file, and produce a graph for it.
The `*.msgb` file is something that we define and ship with the CLI, not supplied by the user.

What would be the case for asking for it as an input arg? When can there be alternative/different `*.msgb` files?

Reply:
As discussed in the stand up, this is useful to have for developing the bindings and for users of slang, to be able to construct arbitrary graphs from a CST, not just the bindings graph. Nevertheless, we decided to the command from the CLI help output in the meantime until we're ready to make it public.

Reply:
I suggest adding a doc comment above the `#[command(hide = true)]` to indicate that it is only for internal development for now.

Reply:
The command is not hidden, but disabled by default and enabled by the feature flag `__graph_builder`. I also added a comment near where the private feature flag is defined to indicate that it's for internal development only.

Reply:
Sounds good.